src/platform/packages/shared/kbn-esql-multiterm-transformer/src/index.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { DatatableColumn, DatatableRow } from '@kbn/expressions-plugin/common';
+
+interface EsqlMultiTermTransformInput {
+  columns: DatatableColumn[];
+  rows: DatatableRow[];
+}
+
+interface EsqlMultiTermTransformOutput {
+  columns: DatatableColumn[];
+  rows: DatatableRow[];
+  transformed: boolean;
+  newColumnName: string | null;
+  originalStringColumns: DatatableColumn[];
+}
+
+/**
+ * Checks a datatable for a specific shape (1 date, 1 number, and 2 or more string columns)
+ * and transforms it if the shape is matched. The transformation combines the string
+ * columns into a single column, creating a unified breakdown dimension for visualizations.
+ *
+ * @param {EsqlMultiTermTransformInput} input - The datatable containing columns and rows.
+ * @returns {EsqlMultiTermTransformOutput} The transformed datatable, or the original if the shape is not matched.
+ */
+export function transformEsqlMultiTermBreakdown({
+  columns,
+  rows,
+}: EsqlMultiTermTransformInput): EsqlMultiTermTransformOutput {
+  if (columns.length <= 2) {
+    return { columns, rows, transformed: false, newColumnName: null, originalStringColumns: [] };
+  }
+
+  // Categorize columns by their data type.
+  const dateColumns = columns.filter((c) => c.meta.type === 'date');
+  const numberColumns = columns.filter((c) => c.meta.type === 'number');
+  const stringColumns = columns.filter((c) => c.meta.type === 'string');
+
+  // Check if the datatable matches the specific shape for transformation.
+  if (dateColumns.length === 1 && numberColumns.length === 1 && stringColumns.length >= 2) {
```

Comment:
what if I want to have 2 metrics to display in the same chart? is limiting a single metric required?

We should also consider that you may want to use a numeric field in the breakdown

Reply:
If the chart has 2 metrics then it skips the transformation and retains the current behavior.